CalendarAppTests/CalendarEventHandlerTests.swift

```diff
@@ -0,0 +1,329 @@
+        XCTAssertEqual(handler.getEventsForDate(tomorrow), [])
+        XCTAssertEqual(handler.getEventsForDate(Date()), [])
+    }
+    
+    func testDeleteMultipleEvents() {
+        let longEvent = SampleEvents.makeEvent(on: yesterday, withDays: 2, withHours: 3)
+        handler.addEventsFromArray([currentEvent, longEvent], Date())
+        handler.addEventsFromArray([earlyEvent, longEvent], yesterday)
+        
+        XCTAssertEqual(handler.getEventsForDate(Date()), [currentEvent, longEvent])
+        XCTAssertEqual(handler.getEventsForDate(yesterday), [earlyEvent, longEvent])
+        
+        handler.deleteEvent(currentEvent)
+        XCTAssertEqual(handler.getEventsForDate(Date()), [longEvent])
+        XCTAssertEqual(handler.getEventsForDate(yesterday), [earlyEvent, longEvent])
+        
+        handler.deleteEvent(longEvent)
+        XCTAssertEqual(handler.getEventsForDate(Date()), [])
+        XCTAssertEqual(handler.getEventsForDate(yesterday), [earlyEvent])
+    }
+    
+    func testAddEventAfterDelete() {
+        let event = Event()
+        handler.addEventsFromArray([currentEvent, event], Date())
+        XCTAssertEqual(handler.getEventsForDate(Date()), [currentEvent, event])
+        handler.deleteEvent(currentEvent)
+        XCTAssertEqual(handler.getEventsForDate(Date()), [event])
+
+        handler.addNewEvent(currentEvent)
+        XCTAssertEqual(handler.getEventsForDate(Date()), [event, currentEvent])
+    }
+    
+    func testUpdateEvent() {
+        handler.addEventsFromArray([currentEvent], Date())
+        let originalStart = currentEvent.startDate
+        let originalEnd = currentEvent.endDate
+        
+        currentEvent.eventTitle = "hi"
+        handler.updateEvent(currentEvent, originalStart, originalEnd)
+        XCTAssertEqual(handler.getEventsForDate(Date()), [currentEvent])
+    }
+    
+    func testUpdateNonexistentEvent() {
+        handler.addEventsFromArray([], Date())
+        let originalStart = currentEvent.startDate
+        let originalEnd = currentEvent.endDate
+        
+        handler.updateEvent(currentEvent, originalStart, originalEnd)
+        XCTAssertEqual(handler.getEventsForDate(Date()), [currentEvent])
+    }
+    
+    func testUpdateEventToNewDate() {
+        handler.addEventsFromArray([currentEvent], Date())
+        handler.addEventsFromArray([], tomorrow)
+        let originalStart = currentEvent.startDate
+        let originalEnd = currentEvent.endDate
+        
+        currentEvent.startDate = currentEvent.startDate.advanced(by: TimeInterval(86400))
+        currentEvent.endDate = currentEvent.endDate.advanced(by: TimeInterval(86400))
+        handler.updateEvent(currentEvent, originalStart, originalEnd)
+        
+        XCTAssertEqual(handler.getEventsForDate(Date()), [])
+        XCTAssertEqual(handler.getEventsForDate(tomorrow), [currentEvent])
+    }
+    
+    func testUpdateEventToNonexistingDate() {
```

Comment:
This model is so hard to reason about. Tomorrow is a date that exists! I sort of think it would be better if the model could handle this scenario but it's excellent work to use the tests to surface that it currently doesn't.